webapp/src/Service/ExternalContestSourceService.php

```diff
@@ -2071,48 +2063,28 @@ protected function addOrUpdateWarning(
             $warning
                 ->setExternalContestSource($this->source)
                 ->setType($type)
-                ->setEntityType($entityType)
+                ->setEntityType($event->type->value)
                 ->setEntityId($entityId);
             $this->em->persist($warning);
         }
 
         $warning
-            ->setLastEventId($eventId)
+            ->setLastEventId($event->id)
             ->setLastTime(Utils::now())
             ->setContent($content);
 
         $this->em->flush();
     }
 
-    protected function removeWarning(string $entityType, ?string $entityId, string $type): void
+    protected function removeWarning(EventType $eventType, ?string $entityId, string $type): void
     {
-        $hash       = ExternalSourceWarning::calculateHash($type, $entityType, $entityId);
-        $warning    = $this->em
+        $hash = ExternalSourceWarning::calculateHash($type, $eventType->value, $entityId);
```

Comment:
Not sure if you touch this in your PR or if it was already broken but calculateHash expects a string for `$entityId` but that can be `null`